Bug: Incompatible signatures when using WebsocketListener class

### Description

`mypy` complains about incompatible signatures when trying to use the `WebsocketListener` class.
The issue can be reproduced by running `mypy` over the documentation example.

`mypy` produces the following error:
```
docs/examples/websockets/listener_class_based_async.py:8: error: Signature of "on_accept" incompatible with supertype "WebsocketListener"  [override]
docs/examples/websockets/listener_class_based_async.py:8: note:      Superclass:
docs/examples/websockets/listener_class_based_async.py:8: note:          Optional[Callable[..., Any]]
docs/examples/websockets/listener_class_based_async.py:8: note:      Subclass:
docs/examples/websockets/listener_class_based_async.py:8: note:          def on_accept(self, socket: WebSocket[Any, Any, Any]) -> Coroutine[Any, Any, None]
docs/examples/websockets/listener_class_based_async.py:11: error: Signature of "on_disconnect" incompatible with supertype "WebsocketListener"  [override]
docs/examples/websockets/listener_class_based_async.py:11: note:      Superclass:
docs/examples/websockets/listener_class_based_async.py:11: note:          Optional[Callable[..., Any]]
docs/examples/websockets/listener_class_based_async.py:11: note:      Subclass:
docs/examples/websockets/listener_class_based_async.py:11: note:          def on_disconnect(self, socket: WebSocket[Any, Any, Any]) -> Coroutine[Any, Any, None]
```

I'm convinced that `docs/examples` should be typed checked, it seems that they are currently not.

### URL to code causing the issue

_No response_

### MCVE

```python
from litestar import Litestar, WebSocket
from litestar.handlers import WebsocketListener


class Handler(WebsocketListener):
    path = "/"

    async def on_accept(self, socket: WebSocket) -> None:
        print("Connection accepted")

    async def on_disconnect(self, socket: WebSocket) -> None:
        print("Connection closed")

    async def on_receive(self, data: str) -> str:
        return data


app = Litestar([Handler])
```


### Steps to reproduce

```bash
1. Run `mypy docs/examples/websockets/listener_class_based_async.py`
```


### Logs

_No response_

### Litestar Version

2.12.1

### Platform

- [X] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

<!-- POLAR PLEDGE BADGE START -->
---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3763">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3763/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3763/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

--- a/litestar/handlers/websocket_handlers/listener.py
+++ b/litestar/handlers/websocket_handlers/listener.py
@@ -335,10 +335,6 @@ class WebsocketListener(ABC):
     """A sequence of :class:`Guard <.types.Guard>` callables."""
     middleware: list[Middleware] | None = None
     """A sequence of :class:`Middleware <.types.Middleware>`."""
-    on_accept: AnyCallable | None = None
-    """Called after a :class:`WebSocket <.connection.WebSocket>` connection has been accepted. Can receive any dependencies"""
-    on_disconnect: AnyCallable | None = None
-    """Called after a :class:`WebSocket <.connection.WebSocket>` connection has been disconnected. Can receive any dependencies"""
     receive_mode: WebSocketMode = "text"
     """:class:`WebSocket <.connection.WebSocket>` mode to receive data in, either ``text`` or ``binary``."""
     send_mode: WebSocketMode = "text"
@@ -380,6 +376,9 @@ class WebsocketListener(ABC):
         self._owner = owner
 
     def to_handler(self) -> WebsocketListenerRouteHandler:
+        on_accept = self.on_accept if self.on_accept != WebsocketListener.on_accept else None
+        on_disconnect = self.on_disconnect if self.on_disconnect != WebsocketListener.on_disconnect else None
+
         handler = WebsocketListenerRouteHandler(
             dependencies=self.dependencies,
             dto=self.dto,
@@ -389,8 +388,8 @@ class WebsocketListener(ABC):
             send_mode=self.send_mode,
             receive_mode=self.receive_mode,
             name=self.name,
-            on_accept=self.on_accept,
-            on_disconnect=self.on_disconnect,
+            on_accept=on_accept,
+            on_disconnect=on_disconnect,
             opt=self.opt,
             path=self.path,
             return_dto=self.return_dto,
@@ -401,6 +400,16 @@ class WebsocketListener(ABC):
         )(self.on_receive)
         handler.owner = self._owner
         return handler
+
+    def on_accept(self, *args: Any, **kwargs: Any) -> Any:
+        """Called after a :class:`WebSocket <.connection.WebSocket>` connection
+        has been accepted. Can receive any dependencies
+        """
+
+    def on_disconnect(self, *args: Any, **kwargs: Any) -> Any:
+        """Called after a :class:`WebSocket <.connection.WebSocket>` connection
+        has been disconnected. Can receive any dependencies
+        """
 
     @abstractmethod
     def on_receive(self, *args: Any, **kwargs: Any) -> Any:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -460,6 +460,7 @@ known-first-party = ["litestar", "tests", "examples"]
 "litestar/_openapi/schema_generation/schema.py" = ["C901"]
 "litestar/exceptions/*.*" = ["N818"]
 "litestar/handlers/**/*.*" = ["N801"]
+"litestar/handlers/websocket_handlers/listener.py" = ["B027"]
 "litestar/params.py" = ["N802"]
 "test_apps/**/*.*" = ["D", "TRY", "EM", "S", "PTH"]
 "tests/**/*.*" = [

--- a/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py
+++ b/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py
@@ -394,10 +394,10 @@ def test_websocket_listener_class_hook_dependencies() -> None:
     class Listener(WebsocketListener):
         path = "/{name: str}"
 
-        def on_accept(self, name: str, state: State, query: dict, some: str) -> None:  # type: ignore[override]
+        def on_accept(self, name: str, state: State, query: dict, some: str) -> None:  # pyright: ignore
             on_accept_mock(name=name, state=state, query=query, some=some)
 
-        def on_disconnect(self, name: str, state: State, query: dict, some: str) -> None:  # type: ignore[override]
+        def on_disconnect(self, name: str, state: State, query: dict, some: str) -> None:  # pyright: ignore
             on_disconnect_mock(name=name, state=state, query=query, some=some)
 
         def on_receive(self, data: bytes) -> None:  # pyright: ignore

